### ISSUE TITLE
Apparent coordinates with get_body inconsistent with JPL Horizons?

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#17300]
I am trying to use ``get_body`` to get the RA/Dec for the sun on a certain date, and am interested in getting the apparent coordinates consistent with JPL horizons. If I go to [Horizons](https://ssd.jpl.nasa.gov/horizons/app.html#/) and choose the following settings:

![image](https://github.com/user-attachments/assets/78eed27e-60ba-43ba-8253-9108e05967b5)

I get the following results:

```
*******************************************************************************
 Date__(UT)__HR:MN     R.A._____(ICRF)_____DEC  R.A.__(a-apparent)__DEC
***********************************************************************
 2024-Nov-01 00:00     14 25 20.09 -14 24 19.7  14 26 39.85 -14 30 57.7
```

For the record here is the geodetic position of the observer as reported by Horizons:

```
Center geodetic : 359.8833, 51.5, 9.9807E-13      {E-lon(deg),Lat(deg),Alt(km)}
```

If I now try and do the same with astropy, I get:

```
In [1]: from astropy.time import Time
   ...: from astropy import units as u
   ...: from astropy.coordinates import EarthLocation, get_body

In [2]: 
   ...: london = EarthLocation.from_geodetic(359.8833 * u.deg, 51.5 * u.deg)
   ...: time = Time("2024-11-01T00:00:00")

In [3]: coord = get_body("sun", time=time, location=london, ephemeris="jpl")
   ...: print(coord.to_string("hmsdms"))
14h25m18.72847303s -14d24m13.08322263s
```

This is close to although not quite the same as the Horizons Astrometric RA/Dec, but quite a bit offset from the Horizons Apparent RA/Dec. What is confusing however is that the docstring for ``get_body`` says that it should be returning apparent coordinates.

If this is user error, it would be nevertheless helpful to indicate in the ``get_body`` docstring how to get results consistent with JPL Horizons since this will be a common use case, and it would also be helpful to know how to select between astrometric versus apparent RA/Dec.

**Comment by astrofrog** [comment]
May be related: https://github.com/astropy/astropy/issues/10230 - is the issue here that the docstring is misleading in calling the coordinates apparent meaning only taking into account light travel time whereas apparent means something different in Horizons?

**Comment by ayshih** [comment]
First, your Astropy code is essentially printing the "apparent" RA/dec oriented to ICRF.  "Apparent" here means that it is correcting for light travel time (via `get_body()`) and for stellar aberration (via `GCRS`).  It is possible – with some effort – to reconstruct the astrometric RA/dec oriented to ICRF for comparison to Horizons, but as you have already seen in #10230, that functionality is not readily available.  For ground observers, the difference between astrometric and apparent RA/dec should be no more than tens of arcseconds.

Second, the "apparent" RA/dec reported by Horizons is _not_ oriented to ICRF.  Here's their definition:
```
 'R.A.__(a-apparent)__DEC' =
  Airless apparent right ascension and declination of the target center
with respect to an instantaneous reference frame defined by the Earth equator
of-date (z-axis) and meridian containing the Earth equinox of-date (x-axis,
EOP-corrected IAU76/80). Compensated for down-leg light-time delay,
gravitational deflection of light, stellar aberration, precession & nutation.
Note: equinox (RA origin) is offset -53 mas from the of-date frame defined
by the IAU06/00a P & N system.
```
That is, we need to compare using a frame other than `GCRS` because we need to include precession and nutation (and technically gravitational deflection of light).  The closest Astropy frame is `PrecessedGeocentric`, which does go beyond `GCRS` in including precession, but there's no native frame that additionally includes nutation.  Extending your example:
```python
In [4]: coord.equinox = time  # Set this frame attribute on SkyCoord to create the equinox-of-date PrecessedGeocentric frame
   ...: print(coord.precessedgeocentric.to_string("hmsdms"))
14h26m40.18189722s -14d30m53.14327595s
```
Note that this is now very close to the Horizons values(`14 26 39.85 -14 30 57.7`), even though nutation is not included in the Astropy values.

As an aside, I'll mention that you're using the default JPL ephemeris for `astropy`, which is DE430, but Horizons is using a newer one.  xref #11609

**Comment by ayshih** [comment]
Here's how to extend your example to compare against the astrometric RA/dec reported from Horizons:
```python
>>> from astropy.coordinates import SkyCoord
>>> obs_in_icrs = SkyCoord(london.get_gcrs(time)).icrs
>>> vector_icrf = coord.icrs.cartesian - obs_in_icrs.cartesian
>>> print(SkyCoord(vector_icrf, frame="icrs").to_string("hmsdms"))
14h25m20.08850928s -14d24m19.7045748s
```
Note that this is spot-on with the values from Horizons at printed precision, even with the different JPL ephemeris.

**Comment by astrofrog** [comment]
Thanks for explaining all this! I feel like it would be really valuable to have some version of this in the docstring, or at least a clarification that 'apparent' does not mean the same as on Horizons.

**Comment by ayshih** [comment]
As corrected by @StuartLittlefair, there _is_ a native frame that includes nutation, `TETE`, so don't use `PrecessedGeocentric` as I suggested above.  Here is the proper comparison of apparent RA/dec:
```python
>>> print(coord.tete.to_string("hmsdms"))
14h26m39.84206426s -14d30m52.39595411s
```
There's still a small disagreement with Horizons (`14 26 39.85 -14 30 57.7`).

**Comment by astrofrog** [comment]
Thank you! Before we close this issue, I think some of this information would be good to have in the docs and especially in the ``get_body`` docstring. If no one else has time, I can try and do it later this week.

**Comment by StuartLittlefair** [comment]
I agree - that would be fantastic, but I am unlikely to find the time to do this. Note that if you do include this in the docs, the slight discrepancy noted by @ayshih is removed if you use exactly the same coordinate frame as Horizons, which is a topocentric frame, e.g `TETE` with the origin set to London, whereas the code above gives TETE with the origin at the geocenter. The code below agrees with Horizons with no discrepancy:

```python
>>> coord.transform_to(TETE(obstime=time, location=london)).to_string("hmsdms")
'14h26m39.85480014s -14d30m57.69164255s'
```

**Comment by ayshih** [comment]
Ha, I really need to check my work before posting.  Thanks for correcting me again, @StuartLittlefair!

**Comment by mkbrewer** [comment]
I agree that calling what is returned by `get_body` an apparent position is incorrect and misleading. I hadn't noticed that in the docstring, but if I had I would have complained about it. What this actually is is the topocentric proper position referred to the equator and equinox at J2000 . This from the USNO glossary: 

**proper place**: direction of an object in the [Geocentric Celestial Reference System (GCRS)](https://aa.usno.navy.mil/faq/asa_glossary#Geocentric-Celestial-Reference-System-(GCRS)) that takes into account orbital or space motion and [light-time](https://aa.usno.navy.mil/faq/asa_glossary#light-time) (as applicable), light deflection, and annual aberration. Thus, the position ([geocentric](https://aa.usno.navy.mil/faq/asa_glossary#geocentric) [right ascension](https://aa.usno.navy.mil/faq/asa_glossary#right-ascension) and [declination](https://aa.usno.navy.mil/faq/asa_glossary#declination)) at which the object would actually be seen from the center of the Earth if the Earth were transparent, non-refracting, and massless. Unless otherwise stated, the coordinates are expressed with respect to the GCRS axes, which are derived from those of the ICRS.

While that definition is for the geocentric proper position, the topocentric proper position is the same with the addition of diurnal parallax and aberration. 

I highly recommend reading USNO Circular 179 if you want to gain a better understanding of all this.

**Comment by mkbrewer** [comment]
In fact, it might be worth considering renaming `_get_apparent_body_position` `_get_proper_body_position`.
